### DISCUSSION TITLE
How to create new db session in websocket ?

### DISCUSSION THREAD

**Discussion by Ahtii** [sqlalchemy/sqlalchemy#5787] [Usage Questions]
Hi,

I have a UI where sender can send a message and the receiver can see old messages and the newly send message but the backend only sends old message and not the newly send message it appear only if i refresh the page which defeats the purpose of websockets here is my code:

```py
def get_db():
    db = database.SessionLocal()
    try:
        yield db
    finally:
        db.close()

@app.websocket("/chat")
async def connect_user(websocket: views.WebSocket, db: Session = Depends(get_db)):
    # setup code for each connection
    user = views.get_current_user(db, websocket.cookies.get("access_token"))
    if user:        
        await socket_manager.connect(websocket, user)               
        try:
            while True:
                data = await websocket.receive_json()                                          
                if data.get('all_messages'): 

                    # how would I create new database session here 
                    # to return all messages  including new ones   
    
                    # code to get the message from database
                else:
                    # code to create message in db
        except WebSocketDisconnect:
            socket_manager.disconnect(websocket, user) 
```

I think the problem is that I am using old db session (once websocket gets connected) that is why i don't get new messages. I checked my database it does get updated with new message but I don't know how would I create a new database session inside websocket. I tried  `db: Session = Depends(get_db)` and `db = get_db()` in the second `if` but it don't work please help.

Tools I am using:

FastAPI
MySQL
Sqlalchemy
Alembic

**Comment by CaselIT** [comment]
Hi,

I'm not really familiar with fastApi, maybe they can better help with your problem.
Regarding creating a session, since I'm guessing you are using v1.4 with async, you can look here https://docs.sqlalchemy.org/en/14/orm/session_basics.html#opening-and-closing-a-session
